examples/plot_how_train_test_and_tune.py

```diff
@@ -0,0 +1,669 @@
+    cv=train_val_split,
+    windows_dataset=windows_dataset,
+    X_train=X_train,
+    y_train=y_train,
+    test_set=test_set,
+)
+######################################################################
+# How to tune your Hyperparameters
+# --------------------------------
+#
+
+######################################################################
+# One way to do hyperparameter tuning is to run each configuration
+# manually (via Option 2 or 3 from above) and compare the validation
+# performance afterwards. In the early stages of your developement
+# process this might be sufficient to get a rough understanding of
+# how your hyperparameter should look like for your model to converge.
+# However, this manual tuning process quickly becomes messy as the
+# number of hyperparameters you want to (jointly) tune increases.
+# Therefore you sould automate this process. We will present two
+# different options, analogous to Option 2 and 3 from above.
+#
+
+######################################################################
+# Option 1: Train-Val-Test Split
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# We will again make use of the sklearn library to do the hyperparameter
+# search. `GridSearchCV <https://scikit-learn.org/stable/modules/
+# generated/sklearn.model_selection.GridSearchCV.html>`__ will perform
+# a Grid Search over the parameters specified in ``param_grid``.
+# We use grid search as a simple example, but you can use `any strategy
+# you want <https://scikit-learn.org/stable/modules/classes.html#
+# module-sklearn.model_selection>`__).
+#
+
+import pandas as pd
+from sklearn.model_selection import GridSearchCV
+
+train_val_split = [
+    tuple(train_test_split(X_train.indices_, test_size=0.2, shuffle=False))
+]
+
+param_grid = {
+    "optimizer__lr": [0.00625, 0.000625],
+}
+search = GridSearchCV(
+    estimator=clf,
+    param_grid=param_grid,
+    cv=train_val_split,
+    return_train_score=True,
+    scoring="accuracy",
+    refit=True,
+    verbose=1,
+    error_score="raise",
+)
+
+search.fit(X_train, y_train, **fit_params)
+search_results = pd.DataFrame(search.cv_results_)
+
+best_run = search_results[search_results["rank_test_score"] == 1].squeeze()
+print(
+    f"Best hyperparameters were {best_run['params']} which gave a validation "
+    f"accuracy of {best_run['mean_test_score'] * 100:.2f}% (training "
+    f"accuracy of {best_run['mean_train_score'] * 100:.2f}%)."
+)
+
+######################################################################
+# Option 2: k-Fold Cross Validation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# To perform a full k-Fold CV just replace ``train_val_split`` from
+# above with the ``KFold`` cross-validator from sklearn.
+
+train_val_split = KFold(n_splits=5, shuffle=False)
+
+######################################################################
+# .. note::
```

Comment:
this is something that should appear at the top and maybe as a warning as it's not a detail.

Reply:
maybe we can use `copy.deepcopy(model)` in every `EEGClassifier` init, so we won't modify the weights of original model?

Reply:
I prefer the warning option. If we use `deepcopy` we would have to include a small note on that as well to avoid confusion.